### ISSUE TITLE
Block Sort is no longer working

### ISSUE THREAD

**Issue by lavermil** [1nVitr0/plugin-vscode-blocksort#33]
Hello again!

I haven't used the extension in a while but when I went to use it I got it to sort once but would no longer sort again. I tried on another machine I had it installed on and it worked the say way. I have tried your examples/etc but nothing seems to get it to work again. I even went back to version 6.x to see that would resolve it.

I have tried without highlighting everything then running just regular block sort vs deep sort.
I have tried highlighting everything then running just regular block sort vs deep sort. 
I have tried having trailing newlines after the last bit of text in the file.
I have tried the sorting of text from saved files and non saved files.
I have tried using you simple switch example and changing around a few things.

All with no success.
Is there any known incompatibility with other extensions?
What is the best way to troubleshoot to see if the extension is firing an event to do something?
Is there a debug mode or something to help troubleshoot this?

Thank you in advance.
lavermil

**Comment by lavermil** [comment]
After even further investigation I think what is happening is the extension gets hung up when doing sorting of the large configs I am doing and then it fails to work until it completes the original sort request. That took well over 5 mins to do.

Does that sound like what the problem might be?

**Comment by 1nVitr0** [comment]
That's possible, I think compatibility with Cancellation Tokens was never implemented, so very time consuming requests might block future requests, although they have actually been cancelled. I will look into it, thanks for the bug report!

**Comment by 1nVitr0** [comment]
@lavermil Can you provide an example file where the sorting failed?

**Comment by lavermil** [comment]
I tried to come up with a config I could share but that was too difficult and then when I randomized something it was too simple and everything worked great. I did however grab an Azure ARM template and try to do Block Deep sort on it and my fan is running high speed while it tries to do it. Here is a link to the Azure ARM template (JSON). Still behaves the same and blocks all future executions for block sort until it finishes.

https://github.com/Azure/azure-quickstart-templates/raw/master/demos/app-service-environment-azfirewall/azuredeploy.json

**Comment by 1nVitr0** [comment]
Thank you @lavermil! I did some testing and realized that the current performance is pretty abysmal.

I implemented line caching instead of getting the same line mutliple times and running very complex Regexes over them every time. This increased the performance in my tests from more than `3 Minutes` for a `60KB` file to less than `300ms`.

The automated tests still run through, so this shouldn't change any of the functionality. Additionally, the sorting now actually aborts when vscode requests it to.

This will be deployed with the next release once the checks run through in the next few minutes. Feel free to report anything that I broke if that is the case.
